packages/autocomplete-preset-algolia/src/search/fetchAlgoliaResults.ts

```diff
@@ -60,6 +77,18 @@ export function fetchAlgoliaResults<TRecord>({
       })
     )
     .then((response) => {
+      // Go through each result and add insights metadata
+      response.results.forEach((result) => {
+        // @ts-ignore
+        (result.hits || result.facetHits).forEach((hit) => {
+          // @ts-ignore
+          hit.__autocomplete_algoliaResultsMetadata = {
+            appId,
+            apiKey,
+          };
+        });
+      });
```

Comment:
Why not map and spread instead of mutating?

Reply:
map/spread would be a better implementation indeed